doc/flow_counters/routes_flow_counters.md

```diff
@@ -0,0 +1,475 @@
+# SONiC Route Flow Counter Design #
+
+## Table of Content
+
+### Revision
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Junchao Chen  | Initial version                   |
+
+### Scope
+
+This document is the design document for route flow counter feature on SONiC.
+
+### Definitions/Abbreviations
+
+N/A
+
+### Overview
+
+Flow counters are usually used for debugging, troubleshooting and performance enhancement processes. Flow counters could cover cases like:
+
+- Routes matching the configured prefix pattern (number of hits and number of bytes)
+- Host interface traps (number of received traps per Trap ID)
+- FDB entries matching the configured VXLAN tunnel or using the VLAN ID as pattern
+- Next-Hop/Next-Hop Group/Next-Hop Group Member
+
+This document focus on route counter.
+
+### Requirements
+
+- Generic Counters shall be used as Flow Counters introduced by the feature
+- CLI shall be used for configuration, showing and clearing of statistics
+- Flow Counters for routes shall be configured using prefix patterns. The flow counter shall be bound to all routes matching the configured pattern (prefix:vrf)
+- In Phase 1 the number of configured route patterns shall be limited to 1 (enforcement shall be done during configuration via CLI)
+- In Phase 1 the number of matching routes shall be limited to the pre-configured value (default value - 30), after reboot it is not ensured that the same set of matching routes will be used for counting
```

Comment:
where is this pre-configured value (30) coming from? what resource is used decides this value?

Reply:
It comes from the previous feature "trap flow counter". We observed that when we enable 30 trap counters and query them every 1 seconds, the CPU usage will increase about 5%. I will do the same test for this feature and update the default value here.